### ISSUE TITLE
Mpirun hangs when running remote command. Seems related to trying to run orted over ssh

### ISSUE THREAD

**Issue by arlininger** [open-mpi/ompi#9115]
### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)
```
$ mpirun --version
mpirun (Open MPI) 2.1.1
```

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

Installed via apt on Ubuntu 18.04. 

### Please describe the system on which you are running

* Operating system/version: Ubuntu 18.04
* Computer hardware: Bare metal with 48 or 64 cores per machine
* Network type: Pair of GigE NICs in a Bond

-----------------------------

## Details of the problem

Using mpirun to run a task on a local system works just fine. However, trying to run it on a remote system will hang. 

```
$ mpirun --mca plm_base_verbose 10 --host atlas -n 1 hostname
[hyperion:29933] mca: base: components_register: registering framework plm components
[hyperion:29933] mca: base: components_register: found loaded component slurm
[hyperion:29933] mca: base: components_register: component slurm register function successful
[hyperion:29933] mca: base: components_register: found loaded component isolated
[hyperion:29933] mca: base: components_register: component isolated has no register or open function
[hyperion:29933] mca: base: components_register: found loaded component rsh
[hyperion:29933] mca: base: components_register: component rsh register function successful
[hyperion:29933] mca: base: components_open: opening plm components
[hyperion:29933] mca: base: components_open: found loaded component slurm
[hyperion:29933] mca: base: components_open: component slurm open function successful
[hyperion:29933] mca: base: components_open: found loaded component isolated
[hyperion:29933] mca: base: components_open: component isolated open function successful
[hyperion:29933] mca: base: components_open: found loaded component rsh
[hyperion:29933] mca: base: components_open: component rsh open function successful
[hyperion:29933] mca:base:select: Auto-selecting plm components
[hyperion:29933] mca:base:select:(  plm) Querying component [slurm]
[hyperion:29933] mca:base:select:(  plm) Querying component [isolated]
[hyperion:29933] mca:base:select:(  plm) Query of component [isolated] set priority to 0
[hyperion:29933] mca:base:select:(  plm) Querying component [rsh]
[hyperion:29933] mca:base:select:(  plm) Query of component [rsh] set priority to 10
[hyperion:29933] mca:base:select:(  plm) Selected component [rsh]
[hyperion:29933] mca: base: close: component slurm closed
[hyperion:29933] mca: base: close: unloading component slurm
[hyperion:29933] mca: base: close: component isolated closed
[hyperion:29933] mca: base: close: unloading component isolated
[hyperion:29933] [[38665,0],0] plm:rsh: final template argv:
        /usr/bin/ssh <template>  orted --hnp-topo-sig 2N:2S:2L3:24L2:24L1:24C:48H:x86_64 -mca ess "env" -mca ess_base_jobid "2533949440" -mca ess_base_vpid "<template>" -mca ess_base_num_procs "2" -mca orte_hnp_uri "2533949440.0;usock;tcp://172.17.22.205:37605" --mca plm_base_verbose "10" -mca plm "rsh" -mca pmix "^s1,s2,cray"
[atlas:26911] mca: base: components_register: registering framework plm components
[atlas:26911] mca: base: components_register: found loaded component rsh
[atlas:26911] mca: base: components_register: component rsh register function successful
[atlas:26911] mca: base: components_open: opening plm components
[atlas:26911] mca: base: components_open: found loaded component rsh
[atlas:26911] mca: base: components_open: component rsh open function successful
[atlas:26911] mca:base:select: Auto-selecting plm components
[atlas:26911] mca:base:select:(  plm) Querying component [rsh]
[atlas:26911] mca:base:select:(  plm) Query of component [rsh] set priority to 10
[atlas:26911] mca:base:select:(  plm) Selected component [rsh]
```

Passwordless ssh is setup and works:

```
$ ssh atlas which orted
/usr/bin/orted
```

However, actually running orted in any respect will hang. Including a snippet of the debug output from ssh:
```
ssh -v atlas orted --help
...
Authenticated to atlas ([<redacted>]:22).
debug1: channel 0: new [client-session]
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: pledge: network
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug1: Sending environment.
debug1: Sending env LANG = en_US.UTF-8
debug1: Sending env LC_CTYPE = en_US.UTF-8
debug1: Sending env LC_TERMINAL = iTerm2
debug1: Sending env LC_TERMINAL_VERSION = 3.4.8
debug1: Sending command: orted --help
```

It seems orted doesn't like being run remotely, not even for the `--help` parameter. However, it doesn't mind being run locally.

**Comment by jsquyres** [comment]
Unfortunately, Open MPI v2.1.1 is pretty ancient, and we haven't supported it for quite a while.

Aside from checking to make sure that all TCP ports are able to be opened between the two hosts, I don't have any other concrete suggestions for the v2.x series.

Can you upgrade to Open MPI v4.1.x?

**Comment by arlininger** [comment]
I went with what was available in the ubuntu repo. I'll see what I can do about upgrading.

**Comment by jsquyres** [comment]
Understood.  Sorry that Ubu 18 is apparently pretty old / out of date.  😦

**Comment by arlininger** [comment]
Got my test machines upgraded to Ubuntu 20.04 which has MPI 4.0.3. I'm having the same basic issue.

```
$ mpirun --mca plm_base_verbose 10 --host atlas -n 1 hostname
[hyperion:05201] mca: base: components_register: registering framework plm components
[hyperion:05201] mca: base: components_register: found loaded component slurm
[hyperion:05201] mca: base: components_register: component slurm register function successful
[hyperion:05201] mca: base: components_register: found loaded component isolated
[hyperion:05201] mca: base: components_register: component isolated has no register or open function
[hyperion:05201] mca: base: components_register: found loaded component rsh
[hyperion:05201] mca: base: components_register: component rsh register function successful
[hyperion:05201] mca: base: components_open: opening plm components
[hyperion:05201] mca: base: components_open: found loaded component slurm
[hyperion:05201] mca: base: components_open: component slurm open function successful
[hyperion:05201] mca: base: components_open: found loaded component isolated
[hyperion:05201] mca: base: components_open: component isolated open function successful
[hyperion:05201] mca: base: components_open: found loaded component rsh
[hyperion:05201] mca: base: components_open: component rsh open function successful
[hyperion:05201] mca:base:select: Auto-selecting plm components
[hyperion:05201] mca:base:select:(  plm) Querying component [slurm]
[hyperion:05201] mca:base:select:(  plm) Querying component [isolated]
[hyperion:05201] mca:base:select:(  plm) Query of component [isolated] set priority to 0
[hyperion:05201] mca:base:select:(  plm) Querying component [rsh]
[hyperion:05201] mca:base:select:(  plm) Query of component [rsh] set priority to 10
[hyperion:05201] mca:base:select:(  plm) Selected component [rsh]
[hyperion:05201] mca: base: close: component slurm closed
[hyperion:05201] mca: base: close: unloading component slurm
[hyperion:05201] mca: base: close: component isolated closed
[hyperion:05201] mca: base: close: unloading component isolated
[hyperion:05201] [[63413,0],0] plm:rsh: final template argv:
        /usr/bin/ssh <template>  orted -mca ess "env" -mca ess_base_jobid "4155834368" -mca ess_base_vpid "<template>" -mca ess_base_num_procs "2" -mca orte_node_regex "hyperion,atlas@0(2)" -mca orte_hnp_uri "4155834368.0;tcp://172.17.22.205:35167" --mca plm_base_verbose "10" -mca plm "rsh" --tree-spawn -mca routed "radix" -mca orte_parent_uri "4155834368.0;tcp://172.17.22.205:35167" -mca pmix "^s1,s2,cray,isolated"
Connection closed by 172.17.22.211 port 22
--------------------------------------------------------------------------
ORTE was unable to reliably start one or more daemons.
This usually is caused by:

* not finding the required libraries and/or binaries on
  one or more nodes. Please check your PATH and LD_LIBRARY_PATH
  settings, or configure OMPI with --enable-orterun-prefix-by-default

* lack of authority to execute on one or more specified nodes.
  Please verify your allocation and authorities.

* the inability to write startup files into /tmp (--tmpdir/orte_tmpdir_base).
  Please check with your sys admin to determine the correct location to use.

*  compilation of the orted with dynamic libraries when static are required
  (e.g., on Cray). Please check your configure cmd line and consider using
  one of the contrib/platform definitions for your system type.

* an inability to create a connection back to mpirun due to a
  lack of common network interfaces and/or no route found between
  them. Please check network connectivity (including firewalls
  and network routing requirements).
--------------------------------------------------------------------------
[hyperion:05201] mca: base: close: component rsh closed
[hyperion:05201] mca: base: close: unloading component rsh
```

**Comment by arlininger** [comment]
Never mind, passwordless ssh broke. Sorry for the noise. I'm closing this until I get through my debugging chain again.
